### ISSUE TITLE
[UX Reliability] HIGH: Investigate and fix Onyx responses applied in incorrect order

### ISSUE THREAD

**Issue by amyevans** [Expensify/App#43788]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

Coming from [this Slack conversation in #wave-collect](https://expensify.slack.com/archives/C036QM0SLJK/p1718283618694479), and originally [this bug report](https://github.com/Expensify/App/pull/43367#issuecomment-2164880222) in [this PR](https://github.com/Expensify/App/pull/43367), calling 2 API commands in quick succession that result in the same Onyx key being `SET` results in the updates being applied in the incorrect order (1st request's result is applied after the 2nd's), resulting in incorrect Onyx data.

## Action Performed:
1. Checkout [this PR](https://github.com/Expensify/App/pull/43367) locally if it has not been merged yet. (This is also likely reproducible with other API commands)
2. As the owner of a Collect or Control policy, visit the /subscription page.
3. Ensure both the `Auto-renew` and `Auto increase` toggles are off
4. Toggle the auto-renew switch on
4. UpdateSubscriptionAutoRenew API command called with autoRenew = true
5. Note this calls the `UpdateSubscriptionAutoRenew` API and returns a response like
```json
{
    "jsonCode": 200,
    "requestID": "89328628ea5fbf4c-WAW",
    "onyxData": [
        {
            "key": "nvp_private_subscription",
            "onyxMethod": "set",
            "value": {
                "addNewUsersAutomatically": false,
                "autoRenew": true,
                "autoRenewLastChangedDate": "2024-06-13 13:45:58",
                "endDate": "2025-06-01",
                "startDate": "2024-06-01",
                "type": "yearly2018"
            }
        }
    ],
    "previousUpdateID": 437784469,
    "lastUpdateID": 437784510
}
```
6. Quickly toggle the auto-increase switch on
7. Notes this calls the `UpdateSubscriptionAddNewUsersAutomatically` API command and returns a response like
8. API returns full nvp_private_subscription Onyx data as SET
```json
{
    "jsonCode": 200,
    "requestID": "8932863338a4b61b-WAW",
    "onyxData": [
        {
            "key": "nvp_private_subscription",
            "onyxMethod": "set",
            "value": {
                "addNewUsersAutomatically": true,
                "autoRenew": true,
                "autoRenewLastChangedDate": "2024-06-13 13:45:58",
                "endDate": "2025-06-01",
                "startDate": "2024-06-01",
                "type": "yearly2018"
            }
        }
    ],
    "previousUpdateID": 437784510,
    "lastUpdateID": 437784512
}
```

## Expected Result:
Since the 2nd API request correctly returned `"addNewUsersAutomatically": true,` and `"autoRenew": true,`, we expect that both toggles are "ON".

## Actual Result:
The responses from the API requests are not applied in the correct order, therefore the auto-increase toggle is "OFF" and only the auto-renew toggle is "ON".

## Screenshots/Videos

https://github.com/Expensify/App/assets/141406735/012d0bf9-c3f1-432d-9eb4-64ab7bf630b4

</details>

## Areas to investigate

1. That the 2 API calls are indeed done sequentially
2. That both events we receive have a previousUpdateID
3. That the IDs are correct
4. Why when events are out of order, we are not correctly realizing that an update is missing and correcting it before applying it to onyx
5. Anything else that comes to mind

cc @JKobrynski @fabioh8010 @iwiznia @DylanDylann 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by alexpensify** [comment]
@amyevans should this be internal or external?

**Comment by amyevans** [comment]
Initial investigation points to it being a frontend bug so let's assume External unless proven otherwise by additional investigation

**Comment by blimpich** [comment]
This is possibly a duplicate of (or related to) https://github.com/Expensify/App/issues/37560. [Relevant slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1718646868832479). We should consider holding this issue until that issue is resolved, which should be soon according to [this update](https://github.com/Expensify/App/issues/37560#issuecomment-2168303115).

**Comment by alexpensify** [comment]
Thanks, @blimpich, for the update! I've updated this GH and moved it to Weekly. I'll keep checking https://github.com/Expensify/App/issues/37560 for updates.

**Comment by alexpensify** [comment]
@blimpich - the other GH closed, should we take the same action here? Thanks!

**Comment by blimpich** [comment]
I don't think so, that issue was closed not because it was fixed but because the contributor decided to stop working on it. Let me look into it for bit, I'd like to have that issue reopened. Otherwise I think we mark this as external.

**Comment by blimpich** [comment]
Hmm okay maybe we do close this. If https://github.com/Expensify/App/issues/37560 and this issue do in fact have the same root cause then it might not be worth working on this. The effort required to complete this vs what we get out of it is not great.

I also don't consider [this proposal](https://github.com/Expensify/App/issues/43788#issuecomment-2171408523) a good solution since it suggests we just prevent the user from toggling the setting. Very hacky solution IMO and doesn't address the root cause.

**Comment by blimpich** [comment]
Yeah after thinking about it for a minute I say we close this as "won't fix." It is very likely too much effort for not much reward. Lets re-visit this if actual users complain about this.

**Comment by alexpensify** [comment]
Thanks for the feedback @blimpich!
